Makefile

```diff
@@ -9,8 +9,8 @@ build install modules upgrade: .build.info
 	cd $(NGINX_SRC_DIR) && $(MAKE) -e $@
 
 clean:
-	cd $(NGINX_SRC_DIR) && $(MAKE) $@
+	-cd $(NGINX_SRC_DIR) && $(MAKE) $@
```

Comment:
this may not work as expected before this patch or after. If there's a `Makefile` in your home directory and `NGINX_SRC_DIR` isn't defined in `.build.info`, the existing code will do a clean there and succeed if that `Makefile` has a clean target that exits with a 0 status code. This is the case because it will just do `cd` (i.e., go to home dir). By adding the dash in this patch, it'll just continue even if there's a `Makefile` there and `clean` fails. If `NGINX_SRC_DIR` is defined and that directory doesn't exist, the `cd` command will fail, but make will carry on (which is probably fine).

To make this more bulletproof though, I think this line should be changed to this:

```makefile	
	test -d "$(NGINX_SRC_DIR)" && $(MAKE) -C $(NGINX_SRC_DIR) $@ || true
```

This has the added benefit of not failing a clean if the `$(NGINX_SRC_DIR)` exists but doesn't have a `Makefile` (e.g., because the configure script hasn't run but the `.build.info` file does exist with a variable defined which points to it).